Source/Core/Core/IOS/ES/TitleManagement.cpp

```diff
@@ -350,7 +350,25 @@ IPCCommandResult ES::ImportContentEnd(Context& context, const IOCtlVRequest& req
 
 ReturnCode ES::ImportTitleDone(Context& context)
 {
-  if (!context.title_import.tmd.IsValid())
+  if (!context.title_import.tmd.IsValid() || context.title_import.content_id != 0xFFFFFFFF)
+    return ES_EINVAL;
+
+  // Make sure all listed, non-optional contents have been imported.
+  const u64 title_id = context.title_import.tmd.GetTitleId();
+  const std::vector<IOS::ES::Content> contents = context.title_import.tmd.GetContents();
+  const IOS::ES::SharedContentMap shared_content_map{Common::FROM_SESSION_ROOT};
+  const bool has_all_required_contents =
+      std::all_of(contents.begin(), contents.end(), [&](const IOS::ES::Content& content) {
```

Comment:
I'm not sure, thats why I ask. The lambda uses a `const Content&`, which is fine, but the non-`const_iterator` doesn't guarantee it. Other than that, I have no idea if this does make a difference in practise (like, performance wise or anything) or whether its just a compile time verification thing like `const` itself.